### PR TITLE
Add scans and certificate metrics to API

### PR DESCRIPTION
**Pull request by autrilla**
We might also be interested in additional metrics such as scans processed, workers analysis executed, and number of trust relationships added, but that would have to run on the scanner, which doesn't have a way to pass arbitrary objects down the call stack like the `http` package allows with `context`.

### CODE REVIEW

File: tlsobs-api/handlers.go [hunk 1]
```diff
@@ -17,8 +17,10 @@ import (
 	"bytes"
 	"crypto/sha256"
 
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
```
Comment (jvehent) [inline tlsobs-api/handlers.go]: Why not put everything into the metrics package and not impact cloudwatch in the api/scanner?

File: tlsobs-api/main.go [hunk 2]
```diff
@@ -71,17 +72,29 @@ func main() {
...
+		setResponseHeaders(),
+	}
+
+	sess, err := session.NewSession()
+	if err != nil {
+		log.Printf("Could not create AWS session: %s", err)
+	} else {
+		cloudwatchSvc := cloudwatch.New(sess)
+		middlewares = append(middlewares, addCloudwatch(cloudwatchSvc))
+		log.Printf("CloudWatch configured.")
```
Comment (jvehent) [inline tlsobs-api/main.go]: This might be cleaner in `middleware.go`.
Also, we should pref it to `off` for people running outside of aws.

Comment (autrilla) [inline tlsobs-api/main.go]: If setting up cloudwatch fails for any reason, we show a warning and keep going. I don't think we need a setting to turn it off if we do this.

**Comment by autrilla** [comment]
I've moved the metric sending primarily to the database package, except where I can't because the scanner performs db queries itself. We should probably make all database interactions go through the database package, but I think it's best to leave that for another PR.

**Pull request merged**